src/Microsoft.OData.Client/Serialization/HttpClientRequestMessage.cs

```diff
@@ -447,7 +412,14 @@ private Task<HttpResponseMessage> CreateSendTask()
             }
 
             _requestMessage.Method = new HttpMethod(_effectiveHttpMethod);
-            return _client.SendAsync(_requestMessage);
+            // If the timeout value is still the default, don't schedule cancellation.
+            // The timeout from the HttpClient will take effect.
+            if (_isTimeoutProvidedByCaller)
+            {

```

Comment:
Add tests tp verify this.

Reply:
I did. See: https://github.com/OData/odata.net/pull/2883/files#diff-b4128975c89acf84c4f440fc4c0b2af782a08e0fb8955d0195ed344bab8562a8R317